File: cwiczenia_tablicowe/cw2_zad3/maclaurin.py

def lnp1(x,eps):
  sum=0
  newsum=x
  i=1
  while abs(sum - newsum) >= eps:
    i+=1
    sum=newsum
    newsum=newsum + (((-1)**(i-1))/i)*(x**i)
  return newsum, i


lnp_08, n_kroków =lnp1(0.8,1e-30)

print(f"mynik : {lnp_08:.6f}, kroków: {n_kroków}")

File: cwiczenia_tablicowe/cw2_zad3/maclaurin.R

lnp1<-function(x, eps){
sum<-0
newsum<-x
i<-1
while( abs(sum - newsum) >= eps){
i<- i +1
sum<- newsum
newsum<- newsum + (((-1)^(i-1))/i)*(x^i)
}
attr(sum, "n_steps")<- i
return(sum)
}


lnp1(x=0.8,eps=1e-30)-> lnp_08


print(
sprintf("wynik:  %.6f, n kroków: %d",
		lnp_08, attr(lnp_08, "n_steps"))
      )
